sdk/core/azure_core/src/options/builders.rs

```diff
@@ -57,9 +57,9 @@ pub trait ClientOptionsBuilder {
 }
 
 /// Methods to set general method options for client-specific [`ClientMethodOptions`].
-pub trait ClientMethodOptionsBuilder {
+pub trait ClientMethodOptionsBuilder<'a> {
     /// Set optional [`Context`] for each client method call.
-    fn with_context<P>(mut self, context: &Context<'_>) -> Self
+    fn with_context<P>(mut self, context: &'a Context) -> Self
```

Comment:
This isn't actually accurate, though. `Context` needs to outlive the options that get built; though, that's a clone and this applies to the internal `options: ClientMethodOptions`. And, really, that only needs to outlive the parent `Context`.

Maybe this is fine for now and I should open a tracking issue to consider if this is even worth it. If we end up using the `Context` for tracing, it's likely every derivative `Context` will copy on write anyway so the `Cow` necessitating the lifetime argument would be pointless.